### PR TITLE
Fix categorical selection

### PR DESCRIPTION
**Pull request by astrofrog**
@JudoWill @ChrisBeaumont - this is an immediate fix for the selection in scatter plots with rectangular selections when categorical ROI are present.

I would like to merge this pretty soon and release 0.5.3, since this is a reasonably big bug, and #692 could then go in 0.6.

### CODE REVIEW

File: glue/clients/scatter_client.py [hunk 1]
```diff
@@ -259,13 +259,14 @@ def _process_categorical_roi(self, roi):
...
+
             for coord, lo, hi in axes:
                 comp = list(self._get_data_components(coord))
                 if comp:
                     if comp[0].categorical:
-                        subset = CategoricalRoiSubsetState.from_range(comp[0], self.xatt, lo, hi)
+                        subset = CategoricalRoiSubsetState.from_range(comp[0], self._get_attribute(coord), lo, hi)
                     else:
-                        subset = RangeSubsetState(lo, hi, self.xatt)
+                        subset = RangeSubsetState(lo, hi, self._get_attribute(coord))
```
Comment (astrofrog) [inline glue/clients/scatter_client.py]: Just for info, this was part of the bug (it was always selecting on x-axis)

File: glue/core/roi.py [hunk 2]
```diff
@@ -1232,7 +1232,18 @@ def from_range(cat_comp, lo, hi):
...
+        # Note that we used ceil for lo, because if lo is 0.9 then we should
+        # only select 1 and above.
+
+        lo = np.ceil(lo) if lo > 0 else 0
+        hi = np.ceil(hi) if hi > 0 else 0
+
         roi = CategoricalRoi()
         cat_data = cat_comp._categories
-        roi.update_categories(cat_data[np.floor(lo):np.ceil(hi)])
+        roi.update_categories(cat_data[lo:hi])
```
Comment (astrofrog) [inline glue/core/roi.py]: And this was another bug - if the ROI went below 0, lo was negative and could end up producing a weird slice.

**Comment by ChrisBeaumont** [comment]
Bugfixes look reasonable to me

**Comment by astrofrog** [comment]
@ChrisBeaumont - thanks for reviewing!

**Pull request merged**